core/src/main/java/org/apache/accumulo/core/clientImpl/TableOperationsImpl.java

```diff
@@ -1017,6 +1019,30 @@ public void setProperty(final String tableName, final String property, final Str
     }
   }
 
+  @Override
+  public void modifyProperties(String tableName, final Consumer<Map<String,String>> mapMutator)
+      throws AccumuloException, AccumuloSecurityException, IllegalArgumentException,
+      ConcurrentModificationException {
+    EXISTING_TABLE_NAME.validate(tableName);
+    checkArgument(mapMutator != null, "mapMutator is null");
+
+    final Map<String,String> properties = ThriftClientTypes.CLIENT.execute(context,
+        client -> client.getTableProperties(TraceUtil.traceInfo(), context.rpcCreds(), tableName));
+    mapMutator.accept(properties);
+
+    try {
+      // Send to server
+      ThriftClientTypes.MANAGER.executeVoid(context, client -> client
+          .modifyTableProperties(TraceUtil.traceInfo(), context.rpcCreds(), tableName, properties));
+
+      for (String property : properties.keySet()) {
+        checkLocalityGroups(tableName, property);
```

Comment:
I need to research this some more and get back to you, so the following comment is just the beginning of a discussion (feel free to ignore now).  Taking a quick look at this, i suspect checkLocalityGroups could be restructured to support setting multiple properties at once.  I'll update with a new comment after I do some more digging.

Reply:
Same as above comment, this could probably be a follow on task as a separate issue and PR as an improvement.